### ISSUE TITLE
Question about Weighted quantile sketch

### ISSUE THREAD

**Issue by buptzcW** [dmlc/xgboost#5822]
hi all，
I just read the paper, and it's hard for me to understand why Weighted quantile sketch algorithm uses h_i as weights. In my view, Weighted quantile sketch is used to find a good splitting point efficiently and a good spliiting point can help to reduce the loss. THE QUESTION is why Weighted quantile sketch algorithm uses h_i as weight can help to reduce the loss. 
![image](https://user-images.githubusercontent.com/33115565/85378112-b1a99d80-b56c-11ea-92eb-0aa0359a0041.png)
The equation above shows that h_i can be used as sample weight, but how to reduce loss using h_i as weight? I can't figure out the relationship between h_i and reducing loss and advances using h_i as sample wieght.

**Comment by thvasilo** [comment]
$h_i$ which is the Hessian for data point i, can be used as a proxy for the importance of each sample in the learning process. As such, we can weigh each data point by its Hessian, in order to focus the learning on these samples that cause the most error.

The weighted quantile sketch can be used when we are proposing split points, by weighting data points by their Hessian value.

Now AFAIK, this is not used practically any more, because the split point suggestions do not take Hessians into account, they are proposed only once at the beginning of learning, based on global statistics (on the entire dataset). The devs can correct me on that though.

**Comment by trivialfis** [comment]
Thanks!.  As explained by @thvasilo the hessian is used as proxy for `approx` tree method and `grow_local_histmaker` updater.  While `hist` and `gpu_hist` don't use it.

> Now AFAIK, this is not used practically any more, because the split point suggestions do not take Hessians into account

Actually I'm working on categorical data support, in order to do so I need to bring these 2 tree methods to modern code base.  Also I think in some datasets using hessian does yield better accuracy.  So might see something interesting.

**Comment by gitfourteen** [comment]
> hi all，
> I just read the paper, and it's hard for me to understand why Weighted quantile sketch algorithm uses h_i as weights. In my view, Weighted quantile sketch is used to find a good splitting point efficiently and a good spliiting point can help to reduce the loss. THE QUESTION is why Weighted quantile sketch algorithm uses h_i as weight can help to reduce the loss.
> ![image](https://user-images.githubusercontent.com/33115565/85378112-b1a99d80-b56c-11ea-92eb-0aa0359a0041.png)
> The equation above shows that h_i can be used as sample weight, but how to reduce loss using h_i as weight? I can't figure out the relationship between h_i and reducing loss and advances using h_i as sample wieght.

Off topic, is the minus in the squared term actually + instead?

**Comment by rp-winter** [comment]
> Off topic, is the minus in the squared term actually + instead?
I too am getting a + sign. I read the paper but I still am not able to find how he got to this equation from equation 3 of the Loss function.

**Comment by huangchaosp** [comment]
> > hi all，
> > I just read the paper, and it's hard for me to understand why Weighted quantile sketch algorithm uses h_i as weights. In my view, Weighted quantile sketch is used to find a good splitting point efficiently and a good spliiting point can help to reduce the loss. THE QUESTION is why Weighted quantile sketch algorithm uses h_i as weight can help to reduce the loss.
> > ![image](https://user-images.githubusercontent.com/33115565/85378112-b1a99d80-b56c-11ea-92eb-0aa0359a0041.png)
> > The equation above shows that h_i can be used as sample weight, but how to reduce loss using h_i as weight? I can't figure out the relationship between h_i and reducing loss and advances using h_i as sample wieght.
> 
> Off topic, is the minus in the squared term actually + instead?

I think the - sign in the squared term should be a + sign.
